File: 0002-add-two-numbers/0002-add-two-numbers.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function (l1, l2) {
    let carry = 0
    let dummy = new ListNode()
    let temp = dummy

    while (l1 || l2 || carry) {
        list1 = l1? l1.val : 0
        list2 = l2? l2.val : 0
        sum = list1 + list2 + carry
        carry = Math.floor(sum / 10)
        sum = sum % 10
        temp.next = new ListNode(sum)
        temp = temp.next
        l1 = l1? l1.next : null
        l2 = l2? l2.next : null
    }
    return dummy.next
};

File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        list1 = l1
        list2 = l2
        carry = 0
        dummy = ListNode()
        temp = dummy
        sums = 0 

        while list1 or list2 or carry:
            first = list1.val if list1 else 0
            second= list2.val if list2 else 0
            sums = first + second + carry # 7 ,10,8
            carry = sums // 10 # 0 , 1,0
            sums = sums % 10 # 7,0,8
            temp.next = ListNode(sums) # 7,0,8
            temp = temp.next
            list1 = list1.next if list1 else None
            list2 = list2.next if list2 else None
        return dummy.next





